File: src/main/java/com/github/rabend/generators/IntegerGenerator.java
package com.github.rabend.generators;

import com.fasterxml.jackson.databind.JsonNode;

import java.util.concurrent.ThreadLocalRandom;

public class IntegerGenerator extends AbstractValueGenerator {
    @Override
    public String generateRandomValue(final JsonNode node) {
        Integer randomInt;
        ThreadLocalRandom random = ThreadLocalRandom.current();

        if (node.has("minimum")) {
            randomInt = random.nextInt(node.get("minimum").asInt(), Integer.MAX_VALUE);
            return randomInt.toString();
        }

        randomInt = random.nextInt(0, Integer.MAX_VALUE);
        return randomInt.toString();
    }
}

File: src/main/java/com/github/rabend/generators/IntegerGenerator.kt
package com.github.rabend.generators

import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.JsonPrimitive
import kotlinx.serialization.json.int
import kotlinx.serialization.json.jsonPrimitive
import java.util.concurrent.ThreadLocalRandom

class IntegerGenerator: AbstractValueGenerator {
    override fun generateRandomValue(baseObject: JsonObject): JsonElement{
        val random = ThreadLocalRandom.current()
        val randomInt = if (baseObject.containsKey("minimum")) {
            random.nextInt(baseObject["minimum"]!!.jsonPrimitive.int, Int.MAX_VALUE)
        } else {
            random.nextInt(0, Int.MAX_VALUE)
        }
        return JsonPrimitive(randomInt)
    }
}
